tmail-backend/jmap/extensions-es-v6/src/main/java/com/linagora/tmail/james/jmap/ES6EmailAddressContactSearchEngine.java

```diff
@@ -128,20 +133,29 @@ public Publisher<EmailAddressContact> autoComplete(AccountId accountId, String p
         SearchRequest request = new SearchRequest(configuration.getUserContactReadAliasName().getValue(), configuration.getDomainContactReadAliasName().getValue())
             .source(new SearchSourceBuilder()
                 .size(limit)
-                .query(QueryBuilders.boolQuery()
-                    .must(QueryBuilders.multiMatchQuery(part, EMAIL, FIRSTNAME, SURNAME))
-                    .should(QueryBuilders.termQuery(ACCOUNT_ID, accountId.getIdentifier()))
-                    .should(QueryBuilders.termQuery(DOMAIN, Username.of(accountId.getIdentifier()).getDomainPart()
-                        .map(Domain::asString)
-                        .orElse("")))
-                    .minimumShouldMatch(1)));
+                .query(buildAutoCompleteQuery(accountId, part)));
 
         return client.search(request, RequestOptions.DEFAULT)
             .flatMapIterable(searchResponse -> ImmutableList.copyOf(searchResponse.getHits().getHits()))
             .map(Throwing.function(this::extractContentFromHit).sneakyThrow())
             .distinct(EmailAddressContact::id);
     }
 
+    private QueryBuilder buildAutoCompleteQuery(AccountId accountId, String part) {
+        String[] mustFields = Optional.of(part.contains("@"))
+                .filter(FunctionalUtils.identityPredicate())
+                .map(mailPart -> EMAIL_SEARCH_FIELDS)
+                .orElse(ALL_SEARCH_FIELDS);
+
+        return QueryBuilders.boolQuery()
+                .must(QueryBuilders.multiMatchQuery(part, mustFields))
+                .should(QueryBuilders.termQuery(ACCOUNT_ID, accountId.getIdentifier()))
```

Comment:
I guess Benoit's suggestion is only to use "term query" when only query against the Email field here.

Reply:
`matchQuery` is more suitable, 
one more point, termQuery does not support insensitive cases search with ES < 7.10